kinbot/mess.py

```diff
@@ -144,127 +167,316 @@ def write_input(self):
-            bimols += bimolec_blocks[bimol] + '\n!****************************************\n'
-        termols = ''
-        for termol in termolec_blocks:
-            termols += termolec_blocks[termol] + '\n!****************************************\n'
-        tss = ''
-        for ts in nts_blocks:
-            tss += nts_blocks[ts] + '\n!****************************************\n'
-        barrierless = ''
-        for rxn in barrierless_blocks:
-            barrierless += barrierless_blocks[rxn] + '\n!****************************************\n'
- 
-        dummy_template = pkg_resources.resource_filename('tpl', 'mess_dummy.tpl')
-        with open(dummy_template) as f:
-            dummy = f.read()
-        dum = dummy.format(barrier='tsd', reactant=self.well_names[self.species.chemid], dummy='d1')
-
-        barrierless_header = pkg_resources.resource_filename('tpl', 'mess_barrierless_header.tpl')
-        with open(barrierless_header) as f:
-            barrierless_tpl = f.read()
-        
-        termol_header = pkg_resources.resource_filename('tpl', 'mess_termol_header.tpl')
-        with open(termol_header) as f:
-            termol_tpl = f.read()
-
-        f_out = open('me/mess.inp', 'w')
-        f_out.write(header + '\n!****************************************\n')
-        f_out.write(wells)
-        f_out.write(bimols)
-        f_out.write(tss)
-        f_out.write(termol_tpl)
-        f_out.write(termols)
-        f_out.write(barrierless_tpl)
-        f_out.write(barrierless)
-        #f_out.write(dum)
-        f_out.write('\n!****************************************\nEnd ! end kinetics\n')
-        f_out.close()
+        uq_iter = 0
+        while (uq_iter < uq_n):
+            fi = open('uq.log', 'a')
+            fi.write("uq iteration {}".format(uq_iter))
+            fi.close()
+            # set UQ factors for each i run
+            if uq_iter == 0:
+                well_add = barrier_add = 0.0
+                freqFactor = imagfreqFactor = 1.0
+
+            well_blocks = {}
+            ts_blocks = {}
+            bimolec_blocks = {}
+            termolec_blocks = {}
+            termolec_ts_blocks = {}
+            barrierless_blocks = {}
+            allTS = {}
+
+            if uq_iter >= 0:
+                # energy and freq for each iteration of UQ
+                # arrays reset at the start of each iteration to hold new values
+                ts_e_iter = []
+                ts_imagFreq_iter = []
+                ts_freq_iter = []
+                ts_rxnName_iter = []
+
+                well_e_iter = []
+                well_fr_iter = []
+
+                prod_e_iter = []
+                prod_fr_iter = []
+                bimol_e_iter = []
+                bimol_fr_iter = []
 
+                termol_e_iter = []
+                termol_fr_iter = []
+
+                barrierless_e_iter = []
+                barrierless_fr_iter = []
+
+                well_uqVal = float(well_uq)
```

Comment:
Is this to allow the user to input integers? It's fine, but I guess we want them to input a float. You could also convert the values in parameters.py right from the beginning - maybe that's better?

Reply:
Yes, I can convert values right in the beginning. But that was the idea. A user might input an integer instead of a float if they want values that are not the default.

Reply:
Yes, converting at the beginning would be better I think.